src/io/data_buffers/partitioned_io_buffer.cpp

```diff
@@ -29,48 +29,97 @@
 
 lbann::partitioned_io_buffer::partitioned_io_buffer(lbann_comm *comm, int num_parallel_readers, std::map<execution_mode, generic_data_reader *> data_readers, int num_child_layers)
   : generic_io_buffer(comm, num_parallel_readers, data_readers),
-    M_local(num_child_layers, nullptr) {}
+    m_num_samples_fetched(0) {
+  if (num_child_layers != 2) { /// @todo Generalize
+    LBANN_ERROR("currently only supported with two child layers");
+  }
+  m_input_buffers.clear();
+  m_input_buffers.resize(num_child_layers);
+  m_input_buffers[0].reset(AbsDistMat::Instantiate(comm->get_model_grid(), 0,
```

Comment:
I wouldn't use `Instantiate` here. That function is for dynamic construction and this should clearly be static. The type you want is `DistMatrix<DataType, El::Dist::STAR, El::Dist::VC, El::DistWrap::Element, El::Device::CPU>` and I statically know that because you typed it. This will save you a terrifying abhorrent `if ... else if ... else if ... else if ... else if ... else if ... else if ... *else if ... * <repeat * to * ad nauseum> else` block.

Reply:
Fixed.